shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -30,10 +30,13 @@
 class ResourceExtractor {
     private static final String TAG = "ResourceExtractor";
     private static final String TIMESTAMP_PREFIX = "res_timestamp-";
+    // Build.SUPPORTED_ABIS is only available in later Android API versions.
+    private static final String[] SUPPORTED_ABIS = Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP ? new String[0] : Build.SUPPORTED_ABIS;
```

Comment:
Instead of nesting the for loop I decided to use a local variable for what ABIS are really "supported". On lower SDKs the list is empty.

Reply:
You probably want something like this:

public static String[] supportedAbis() {
    if (Build.VERSION.SDK_INT >= 21 && Build.SUPPORTED_ABIS.length > 0) {
        return Build.SUPPORTED_ABIS;
    } else if (Build.CPU_ABI2 != null && Build.CPU_ABI2.length > 0) {
        return new String[] {Build.CPU_ABI, Build.CPU_ABI2};
    } else {
        return new String[] {Build.CPU_ABI};
    }
}

On lower versions you're supposed to use Build.CPU_ABI and Build.CPU_ABI2:
https://developer.android.com/reference/android/os/Build.html#CPU_ABI

Reply:
Ah got it. Updated, thanks!